src/Sentry/SentrySdk.cs

```diff
@@ -626,29 +657,47 @@ public static void CauseCrash(CrashType crashType)
                 break;
 
 #if ANDROID
-                case CrashType.Java:
-                    JavaSdk.Android.Supplemental.Buggy.ThrowRuntimeException(msg);
-                    break;
+            case CrashType.Java:
+                JavaSdk.Android.Supplemental.Buggy.ThrowRuntimeException(msg);
+                break;
 
-                case CrashType.JavaBackgroundThread:
-                    JavaSdk.Android.Supplemental.Buggy.ThrowRuntimeExceptionOnBackgroundThread(msg);
-                    break;
+            case CrashType.JavaBackgroundThread:
+                JavaSdk.Android.Supplemental.Buggy.ThrowRuntimeExceptionOnBackgroundThread(msg);
+                break;
 
-                case CrashType.Native:
-                    NativeCrash();
-                    break;
+            case CrashType.Native:
+                NativeCrash();
+                break;
 #elif __IOS__
             case CrashType.Native:
                 SentryCocoaSdk.Crash();
                 break;
+#elif NET8_0_OR_GREATER
+            case CrashType.Native:
+                if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                {
+                    NativeStrlenMSVCRT(IntPtr.Zero);
+                }
+                else
+                {
+                    NativeStrlenLibC(IntPtr.Zero);
```

Comment:
should we compile the `.c` lib for Windows/macOS/Linux and have a crash that's also explicit "Crashed on purpose because ABC" ?

It's also nice to see the line number of the native function in the crash report (we also test that we upload debug files to make it work)

Reply:
I'm not sure that having these is worth all the work needed to make it work:
* c code
* compilation in CI and locally
* packaging
* linking for native aot

Reply:
If anything, I'd say let's try to get sentry-native to include that code

Reply:
that's a good idea, just add a Crash function to `sentry-native` like we have on `sentry-cocoa`

Reply:
https://github.com/getsentry/sentry-native/issues/913